### DISCUSSION TITLE
Creating "!reference" tag for array

### DISCUSSION THREAD

**Discussion by robross0606** [mikefarah/yq#2116] [Q&A] [answered]
I need to output some YAML from scratch that looks like this:
```yaml
  rules:
    - !reference [.artifact-rules, rules]
````

I tried using the `tag` command with a custom tag but this doesn't seem to work as expected:
```
  yq -n '
  {
    "rules": [
      (["artifact-rules", "rules"] tag="!reference")
    ]
  }
  '
```
Attempting to tag the array results in wiping out the contents of the array for reasons I don't understand:
```yaml
rules:
  -
```
However, running without the tag produces the content, though not in the array style I would like:
```yaml
rules:
  - - artifact-rules
    - rules
```
Is any of this possible with `yq`?

**Comment by robross0606** [comment]
I think I might have figured out at least one way to do it:
```
  yq -n '
  {
    "rules": [
      (["artifact-rules", "rules"] | . tag="!reference")
    ]
  }
  '
```
Yields
```yaml
  rules:
    - !reference
      - .artifact-rules
      - rules
```

Still not sure if there's some way to change the array style.

**Reply by mikefarah** [reply, marked as answer]
You can use the style tag to change the array style:

```
yq -n '.rules = [ [".artifact-rules", "rules"] | . tag = "!reference" | . style= "flow" ]'
rules:
  - !reference [.artifact-rules, rules]
```
